nx/lib/nx/binary_backend/matrix.ex

```diff
@@ -952,7 +971,7 @@ defmodule Nx.BinaryBackend.Matrix do
   end
 
   defp approximate_zeros(matrix, tol) do
-    do_round = fn x -> if abs(x) < tol, do: 0, else: x end
+    do_round = fn x -> if Complex.abs(x) < tol, do: 0, else: x end
```

Comment:
```suggestion
    do_round = fn x -> if Complex.abs(x) < tol, do: 0 * x, else: x end
```

to ensure that if x is complex, it stays complex